pandas/io/common.py

```diff
@@ -976,11 +971,30 @@ def __next__(self) -> str:
         return newline.lstrip("\n")
 
 
-# Wrapper that wraps a StringIO buffer and reads bytes from it
-# Created for compat with pyarrow read_csv
-class BytesIOWrapper(io.BytesIO):
-    buffer: StringIO | TextIOBase | None
+class _IOWrapper:
+    # TextIOWrapper is overly strict: it request that the buffer has seekable, readable,
+    # and writable. If we have a read-only buffer, we shouldn't need writable and vice
+    # versa. Some buffers, are seek/read/writ-able but they do not have the "-able"
+    # methods, e.g., tempfile.SpooledTemporaryFile.
+    # If a buffer does not have the above "-able" methods, we simple assume they are
+    # seek/read/writ-able.
+    def __init__(self, buffer: BaseBuffer):
+        self.buffer = buffer
+        self.attributes = tuple(
```

Comment:
can you make these separate properties and just do the check of the buffer inside them? its a bit longer code but should be more obvious what is going on

Reply:
Okay, changed it. I made them methods and not properties, as `typing.IO` defines them as methods:
https://github.com/python/typeshed/blob/1d5857e1e0e0fe6078e563d769f07fe3a3352cc9/stdlib/typing.pyi#L522

Reply:
oh sure